File: TestRestApi/src/main/java/com/amuyu/testrestapi/network/RestApiService.java
package com.amuyu.testrestapi.network;

import java.util.Map;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Response;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;
import rx.Observable;

/**
 * Created by noco on 2016-10-12.
 */
public interface RestApiService {


    @GET
    Observable<Response<ResponseBody>> callApi(
            @Url String path, @QueryMap Map<String, String> options
    );

    @POST
    Observable<Response<ResponseBody>> callApiPost(
            @Url String path, @Body RequestBody body
    );
}

File: TestRestApi/src/main/java/com/amuyu/testrestapi/network/RestApiService.kt
package com.amuyu.testrestapi.network

import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.Response
import retrofit2.http.*
import rx.Observable

/**
 * Created by noco on 2016-10-12.
 */
interface RestApiService {


    @GET
    fun callApi(
            @Url path: String, @QueryMap options: Map<String, String>
    ): Observable<Response<ResponseBody>>

    @POST
    fun callApiPost(
            @Url path: String, @Body body: RequestBody
    ): Observable<Response<ResponseBody>>
}
